src/TConsole.cpp

```diff
@@ -2348,3 +2348,33 @@ void TConsole::slot_clearSearchResults()
     mUpperPane->forceUpdate();
     mLowerPane->forceUpdate();
 }
+
+void TConsole::checkNumberOfLines(const int lineCount)
```

Comment:
I think this is hitting https://www.linguistic-antipatterns.com/?tab=%22Unexpected-side-effects%22, it says "check" but it doesn't just check but also raises an event. 

Just throwing an idea out there so I just don't have critique only, how about something like this:

if (exceededConsoleHeight(lineCount)) {
  raiseOverflowEvent();
}

Reply:
I was trying to avoid something that was too long `checkAndRaiseEventOnExcessLinesInNonScrollingConsole(...)`!

As you can see it tries to find a reason ***not*** to raise the event by checking progressively more "involved" things until it has eliminated them all...

Reply:
How about `raiseLinesOverflowEventifRequired(...)`

Reply:
Hmm I was thinking of splitting up the checking and raising also for a separation of concerns reason... but if we're sticking with one function, how about something more concise like `handleLinesOverflowEvent()`?

Reply:
Okay I'll change it (again) to that...